scripts/camera/base-camera.js

```diff
@@ -1,82 +1,77 @@
-import { Script, Vec2, Vec3, math } from 'playcanvas';
+import { Entity, Script, Vec3, Vec2, math } from 'playcanvas';
+
+/** @import { CameraComponent } from 'playcanvas' */
 
 const LOOK_MAX_ANGLE = 90;
 
 class BaseCamera extends Script {
     /**
-     * @type {Entity}
-     */
-    entity;
-
-    /**
-     * @type {HTMLElement}
+     * @type {CameraComponent}
+     * @protected
      */
-    target = document.documentElement;
+    _camera = null;
 
     /**
-     * @attribute
-     * @type {number}
+     * @type {Vec3}
+     * @protected
      */
-    sceneSize = 100;
+    _origin = new Vec3(0, 1, 0);
 
     /**
-     * @attribute
-     * @type {number}
+     * @type {Vec3}
+     * @protected
      */
-    lookSensitivity = 0.2;
+    _position = new Vec3();
 
     /**
-     * @attribute
-     * @type {number}
+     * @type {Vec2}
```

Comment:
Could we add one line description to these attributes. They'll be picked up by the editor as tooltip descriptions